src/flood.rs

```diff
@@ -1,78 +1,191 @@
 use crate::view::layer::LayerCoords;
-use crate::view::{Snapshot, View, ViewResource};
+use crate::view::{View, ViewResource};
 use rgx::color::Rgba8;
 use rgx::kit::shape2d::{Fill, Rotation, Shape, Stroke};
 use rgx::kit::ZDepth;
+use rgx::math::Point2;
 use rgx::rect::Rect;
 
-pub fn flood_fill(
-    view: &View<ViewResource>,
-    starting_point: LayerCoords<u32>,
+struct Grid {
+    pixels: Vec<Rgba8>,
+    pub width: usize,
+    pub height: usize,
+}
+
+impl Grid {
+    pub fn new(pixels: Vec<Rgba8>, width: usize, height: usize) -> Grid {
+        Grid {
+            pixels,
+            width,
+            height,
+        }
+    }
+
+    pub fn get(&self, x: usize, y: usize) -> Option<&Rgba8> {
+        if x < self.width && y < self.height {
+            self.pixels.get(x + y * self.width)
+        } else {
+            None
+        }
+    }
+
+    pub fn get_mut(&mut self, x: usize, y: usize) -> Option<&mut Rgba8> {
+        if x < self.width && y < self.height {
+            self.pixels.get_mut(x + y * self.width)
+        } else {
+            None
+        }
+    }
+}
```

Comment:
Could https://github.com/cloudhead/rx/blob/master/src/pixels.rs#L26 be re-used here? Or does the underlying buffer need to be owned?

Reply:
Actually even in the latter case, you can create a `PixelsMut` out of the owned vector probably?

Reply:
`PixelsMut` looks pretty close to what we need, but the fact that it doesn't own the data is causing some issues. I don't think I can store a `PixelsMut` on the `FloodFIller`, since I'd need to return a reference to the cloned buffer from `::new`... Not sure if there's a clever way to make the borrow checker happy here.

Reply:
Ah right. I think I have an idea on how to get around that, but this isn't a blocker.